int32 input_dim
int32 output_dim
float64[] var
float64[] field
float64[] gradient
float64[] Qgradient
float64[] hessian
float64[] jacobian
float64[] list_jacobians

